core-api/src/main/java/com/optimizely/ab/config/parser/ConditionJacksonDeserializer.java

```diff
@@ -48,20 +48,37 @@ public ConditionJacksonDeserializer() {
     @Override
     public Condition deserialize(JsonParser parser, DeserializationContext context) throws IOException {
         JsonNode node = parser.getCodec().readTree(parser);
-
-        String conditionsJson = node.textValue();
-        JsonNode conditionsTree = objectMapper.readTree(conditionsJson);
+        JsonNode conditionsTree = node;
+        if (node.isTextual()) {
+            String conditionsJson = node.textValue();
+            conditionsTree = objectMapper.readTree(conditionsJson);
+        }
         Condition conditions = ConditionJacksonDeserializer.parseConditions(objectMapper, conditionsTree);
 
         return conditions;
     }
 
     protected static Condition parseConditions(ObjectMapper objectMapper, JsonNode conditionNode) throws JsonProcessingException {
+
+        if (conditionNode.size() == 0) {
+            return new EmptyCondition();
+        }
+
         List<Condition> conditions = new ArrayList<>();
+        int startingParsingIndex = 0;
         JsonNode opNode = conditionNode.get(0);
         String operand = opNode.asText();
-
-        for (int i = 1; i < conditionNode.size(); i++) {
+        switch (operand) {
+            case "or":
+            case "and":
+            case "not":
+                startingParsingIndex = 1;
+                break;
+            default:
+                operand = "or";
+                break;
+        }
```

Comment:
I like how you extracted an `operand` helper function in the other parser(s)! Maybe do the same here? And maybe even put it in a shared module / define it once, if the types allow?